client/src/www/app/app.ts

```diff
@@ -880,8 +879,29 @@ export class App {
     return !('cordova' in window);
   }
 
-  private setAppTheme(event: CustomEvent) {
-    const theme = event.detail.themePreference;
-    this.themeManager.setThemePreference(theme as ThemePreference);
+  private setAppearance(appearance: Appearance) {
+    const documentClassList = window.document.documentElement.classList;
+    const isSystemDark = matchMedia('(prefers-color-scheme: dark)').matches;
+
+    let applyDarkTheme;
+
+    if (appearance === Appearance.DARK) {
+      applyDarkTheme = true;
+    } else if (appearance === Appearance.LIGHT) {
+      applyDarkTheme = false;
+    } else {
+      // Theme.SYSTEM or any other default
+      applyDarkTheme = isSystemDark;
+    }
+
+    if (applyDarkTheme) {
+      this.rootEl.darkMode = true;
+      documentClassList.add('dark');
+    } else {
+      this.rootEl.darkMode = false;
+      documentClassList.remove('dark');
+    }
+
+    this.rootEl.selectedAppearance = appearance;
```

Comment:
Might need to guard against invalid values read from the settings (like `undefined`). An easy way is to add `appearance = Apperance.SYSTEM` to the `else` block above.